app/controllers/posts_controller.rb

```diff
@@ -1,10 +1,29 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.last_3_posts.includes(:comments)
+    @posts = @user.posts.includes(:comments).order(created_at: :desc)
+  end
+
+  def new
+    @post = Post.new
   end
 
   def show
     @post = Post.includes(:comments).find(params[:id])
   end
+
+  def create
+    @post = Post.new(post_params)
+    if @post.save
+      redirect_to @post.author
+    else
+      render :new, status: :unprocessable_entity
+    end
```

Comment:
- [ ] It will be great to ensure that the post is created on behalf of the currently logged-in user.  Although your implementation will still create a post, You have already defined the `current_user` in the application controller. So you should use that user instead of fetching the `author_id ` from the params. Kindly check the sample below ⬇️ 
```ruby
 @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
```
```ruby
private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```

Reply:
Thanks @peterrobert. I have changed it.